src/mds/MDSRank.cc

```diff
@@ -2949,9 +2957,17 @@ void MDSRankDispatcher::handle_asok_command(
   } else if (command == "flush_path") {
     string path;
     cmd_getval(cmdmap, "path", path);
-    command_flush_path(f, path);
+
+    std::lock_guard l(mds_lock);
+    mdcache->flush_dentry(path, new AsyncResponse(f, std::move(on_finish)));
+    return;
   } else if (command == "flush journal") {
-    command_flush_journal(f);
+    auto respond = new AsyncResponse(f, std::move(on_finish));
+    C_Flush_Journal* flush_journal = new C_Flush_Journal(mdcache, mdlog, this, &respond->css, respond);
+
+    std::lock_guard locker(mds_lock);
```

Comment:
We are always suggested to use the `scoped_lock` instead. For example in https://github.com/ceph/ceph/pull/36221#discussion_r463856968.

Reply:
My research didn't explain why, would you please elaborate?
For a single lock case `lock_guard` is still good: https://stackoverflow.com/a/60172828/5171225

Reply:
Yes, it's my fault proliferating that. I had recently seen that discussion comment too. I would consider `lock_guard` to be a good default choice going forward.

Reply:
I couldn't remember the another discussing PR about this. Since it was @batrick's idea then I am okay with it.